src/core/account/custom-tokens.js

```diff
@@ -91,3 +145,16 @@ export async function loadBuiltinTokens(
     })
   )
 }
+
+function findEngine(ai: ApiInput, pluginId: string): EdgeCurrencyEngine | void {
+  for (const walletId of Object.keys(ai.props.state.currency.wallets)) {
+    const walletOutput = ai.props.output.currency.wallets[walletId]
+    if (
+      walletOutput != null &&
+      walletOutput.engine != null &&
+      ai.props.state.currency.wallets[walletId].pluginId === pluginId
```

Comment:
If you use `const [walletId, wallet] = Object.entries(...)` in the loop you can simplify this condition to:

```js
      wallet.pluginId === pluginId
```

Reply:
I don't think we we can use `Object.entries` in the core. Old Androids don't have it, and we don't have a polyfill.